src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1662,14 +1662,34 @@ public InputStream downloadStream(String fileName) throws SnowflakeSQLException
 
     remoteLocation remoteLocation = extractLocationAndPath(stageInfo.getLocation());
 
-    String stageFilePath = fileName;
+    // when downloading files as stream there should be only one file in source files
+    String sourceLocation =
```

Comment:
I've imagined that the path resolution logic is going to be the same (reused/referenced) in both downloadFile & downloadStream. Not reimplemented.

Reply:
Nevertheless, this change fixes the problem. Tested on aws, gcp & azure

Reply:
Get file is using only srcFiles and here in the API we have fileName paramtere that we need to keep for backward compatibility and check it with srcFiles.
I think we can keep it separated for now